lib/deploy/deploy-support.js

```diff
@@ -795,6 +800,58 @@ async function makeInvocationRole(serviceName, functionName, triggerType) {
   return;
 }
 
+const runtimeTypeMapping = {
+  'nodejs6': 'node_modules',
+  'nodejs8': 'node_modules',
+  'python2.7': ['.egg-info', '.dist-info', '.fun'],
+  'python3': ['.egg-info', '.dist-info', '.fun'],
+  'php7.2': 'extension'
+};
+
+async function detectLibrary(codeUri, runtime, baseDir, wrap = '') {
+  const stats = await lstat(codeUri);
+  if (stats.isFile()) {
+    let featureLibrary = runtimeTypeMapping[runtime];
+
+    const allPath = path.join(baseDir, codeUri);
+    const dirName = path.dirname(allPath);
+    const childDir = await readdir(dirName);
+
+    if (Array.isArray(featureLibrary)) {
+      let flag = false;
+      for (const iterator of featureLibrary) {
+        for (const name of childDir) {
+          if (_.endsWith(name, iterator)) {
+            console.warn(red(`${wrap}Do you forget to place the ${name}'s directory in the CodeUri corresponding to template.yml.`));
```

Comment:
Fun detected that the library directory ``  is not included in function `` CodeUri. Please make sure if it is the right configuration. If yes, please ignore.